File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        Stack<int[]> st = new Stack<>();
        int [] res = new int[n];
        
        for(int i = 0; i < n; i++){
            int temp = temperatures[i];
            while(!st.isEmpty() && temp > st.peek()[0]){
                int[] cur = st.pop();
                res[cur[1]] = i - cur[1];
            }
            st.push(new int[]{temp, i});
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxi = 0
        l, r = 0, 0
        temp = ""

        while r < len(s):
            if s[r] not in temp:
                temp += s[r]
                maxi = max(maxi, len(temp))
                r += 1
            else:
                temp = temp[1:]
                l += 1
        
        return maxi
